util/helpers.go

```diff
@@ -13,14 +14,15 @@ func Pointer[T any](t T) *T {
 }
 
 // CreateLinodeAPIFilter converts variables to API filter string
-func CreateLinodeAPIFilter(label string, tags []string) string {
+// Prefer filter in the order: ID, Label and then tags
+func CreateLinodeAPIFilter(id int, label string, tags []string) string {
 	filter := map[string]string{}
-
-	if label != "" {
+	switch {
+	case id != 0:
+		filter["id"] = strconv.Itoa(id)
+	case label != "":
 		filter["label"] = label
-	}
-
-	if len(tags) != 0 {
+	case len(tags) != 0:
 		filter["tags"] = strings.Join(tags, ",")
 	}
```

Comment:
Using the switch will only ever result in the execution of the first case expression that evaluates to true; if `id != 0` and `len(tags) != 0`, only the `id != 0` case block would be executed. Is this the intended behaviour?

Reply:
id's and labels are both completely unique so if you specify a label there isn't a point of also filtering by tags so the goal is if you specify ID just use that, if there isn't an ID just use label, and if neither of them then fall back to filtering by tags

Reply:
tried to capture this on line 17 but maybe I can make that message more clear that each option is used exclusively

Reply:
Sounds good!

As a follow-up, if `id == 0`, `label == ""` and `len(tags) == 0`, the JSON filter string will still be constructed, but with nothing in it. Is this okay? Will it have any unintended effects when calling the Linode API? Could the function short-circuit, and return an empty string if no filters were set?

Reply:
> each option is used exclusively

:thinking: Should this be three, separate functions perhaps? It would help make the intent clearer when the caller wants to use filtering.

Reply:
it is likely this logic is going to be something all of the controllers will need with "if we have an ID then find the resource by an ID, if we just have a label try and find it by the label, otherwise search for it by the labels" separating the functions might mean we just move that duplicate logic to each of the controllers?

Reply:
I was also considering maybe setting tags as the default so we always have something, we shouldn't be creating resources without any tags or known labels or anything so it might be safe to do that? the default would be to fail anyways because it would not filter anything and the next check would fail because more than 1 resource is returned

Reply:
Just to capture a discussion that happened out of band of this PR:

We floated the idea of using a `type Filter struct` instead of a function, that implements the `encoding/json.Marshaler` and `fmt.Stringer` interfaces instead, with full documentation on the preferences of each of the fields in the `Filter` struct.